pkg/reconciler/committer/committer.go

```diff
@@ -110,6 +110,7 @@ func generatePatchAndSubResources[Sp any, St any](old, obj *Resource[Sp, St]) ([
 
 	// Simultaneous updates of spec and status are never allowed.
 	if specOrObjectMetaChanged && statusChanged {
+		// TODO(panic): remove panic and handle the error
```

Comment:
This can't be handled at runtime - it needs to be handled at code-writing-time, which is what "programmer error" is intended to communicate - perhaps we need to word that better?

Reply:
Ack - my concern was opening up a DOS vector in the case of a rarely used and untested controller code path, but `runtime.HandleCrash` when the controller starts should protect a prod system in that case.